### DISCUSSION TITLE
How do I add the media browser into HA similar to the screenshots on the docs?

### DISCUSSION THREAD

**Discussion by hagak** [blakeblackshear/frigate#637] [General]
I installed the HA integration which is great no more creating all those sensors by hand.  However how do you get that media browser in HA that you have screenshots of?

Thanks

**Comment by paularmstrong** [comment]
You need a `media_source:` in your config (or `default_config:`) ([docs](https://www.home-assistant.io/integrations/media_source/))

After that, it's a question of whether you're using the Frigate addon or running it separately. If you're using the addon, it will show up automatically. If you're running separately, you'll need to set up HA to know about the Frigate media directory somehow.

**Reply by blakeblackshear** [reply]
That's not true. The the media browser does not require homeassistant to have access to the disk where clips are stored. They are served over the network.

**Reply by hagak** [reply]
Ah so I am not using the add on so will have to figure out the step of telling HA about it. The integration does not do that piece.

**Reply by paularmstrong** [reply]
> That's not true. The the media browser does not require homeassistant to have access to the disk where clips are stored. They are served over the network.

I'm confused. What's not true? HA will have to be configured to point to the network share in order for the clips to be shown in the HA media browser if Frigate is run on another system.

**Reply by hagak** [reply]
Looks like this is what i need to do:
https://www.home-assistant.io/more-info/local-media/setup-media/

**Reply by blakeblackshear** [reply]
> > That's not true. The the media browser does not require homeassistant to have access to the disk where clips are stored. They are served over the network.
> 
> I'm confused. What's not true? HA will have to be configured to point to the network share in order for the clips to be shown in the HA media browser if Frigate is run on another system.

It does not need access to a network share. Frigate serves up the clips and recordings with nginx, so you just have to point the integration at frigate. No network share required.

**Reply by hagak** [reply]
so i am confused to then, if it is not a network share then how do point HA to it so that the media browser works?

**Reply by blakeblackshear** [reply]
The term "network share" typically refers to sharing a folder over the network with SMB or NFS. Out of the box, HomeAssistant can serve content from a local disk (or network share) in the media browser. The Frigate integration implements a custom implementation of the media browser which makes it look like you are browsing folders, but you are not. It is querying frigate's api's dynamically and then proxying clips and recordings directly from Frigate. This means you can run frigate anywhere and all you need to do is point the integration at http://frigate-ip:5000 and the media browser works. No need to share a folder over your network.

**Reply by paularmstrong** [reply]
🤦🏻  Totally forgot about that! Maybe worth expanding the Media Browser section in the docs

**Reply by hagak** [reply]
I just needed to add media_source: to my HA configuration.yaml
